src/cpp/src/speculative_decoding/speculative_decoding_impl.cpp

```diff
@@ -162,7 +162,6 @@ ContinuousBatchingPipeline::SpeculativeDecodingImpl::generate(const std::vector<
     ManualTimer generate_timer("speculative_decoding: generate()");
     generate_timer.start();
     OPENVINO_ASSERT(!has_non_finished_requests(), "Generate cannot be called while ContinuousBatchingPipeline is already in running state. Use ContinuousBatchingPipeline::add_request");
-    OPENVINO_ASSERT(input_ids.size() == sampling_params.size());
```

Comment:
in general, this check is needed to compare number of prompts (tokenized prompts in this case) and number of passed generation configs

Reply:
We generate 4 generation config results.. @Wovchena Is it possible to make `.reshape()` in the sample in case of `input_ids.size() < sampling_params.size()` instead of this change?